src/Vector3.h

```diff
@@ -235,6 +235,12 @@ class Vector3
 		return *this;
 	}
 
+	/** Provides a hash of a vector's contents */
+	size_t operator()(const Vector3<T> & a_Vector) const
```

Comment:
Because why should operator() return the has of a vector? It doesn't make sense. The std lib API is designed so that the hash object should be a seperate object from the data object, so that std::hash works.